walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,57 @@
 
 package com.google.sps;
 
+import java.util.Comparator;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      Collection<TimeRange> queryResult = new ArrayList<>();
+      List<Event> eventsList = new ArrayList<>(events);
+      Collections.sort(eventsList, new Comparator<Event>() {
+        public int compare (Event e1, Event e2) {
+            return TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen());
+          }
+        });
+      eventsList.removeIf(e -> (
+        Collections.disjoint(e.getAttendees(), request.getAttendees()) || e.getWhen().duration() <= 0));
```

Comment:
change the order of the ||.  Put the cheapest operations first so they can potentially skip the expensive operations (disjoint being the expensive operation).

nit: static import Collections.disjoint so you only have to type "disjoint"

Reply:
Fixed.